### DISCUSSION TITLE
Request: Add support for expo autolinking

### DISCUSSION THREAD

**Discussion by LeunensMichiel** [react-native-maps/react-native-maps#5290] [Feature requests]
https://docs.expo.dev/modules/autolinking/#what-do-i-need-to-have-in-my-module-to-make-it-autolinkable

I often experience issues in my monorepo that through a combination of this library and @expo/config-plugins that my installation of this library is corrupt (it resolves in an absolute path instead of a relative one)

https://github.com/expo/expo/blob/d86d8052f4365cb034217189c449917619ef8c24/packages/%40expo/config-plugins/src/ios/Maps.ts#L109

I think expo autolink might resolve these issues. For now we fixed it by adding this patch

```diff
diff --git a/build/ios/Maps.js b/build/ios/Maps.js
index 941361a3238d233a458c3ac2264aec9a2601e1bc..6f9149f0bbd3102a553d319a2deb961cc21a2e1c 100644
--- a/build/ios/Maps.js
+++ b/build/ios/Maps.js
@@ -123,7 +123,7 @@ function addMapsCocoaPods(src) {
   return (0, _generateCode().mergeContents)({
     tag: 'react-native-maps',
     src,
-    newSrc: `  pod 'react-native-google-maps', path: File.dirname(\`node --print "require.resolve('react-native-maps/package.json')"\`)`,
+    newSrc: `  pod 'react-native-google-maps', path: (Pathname.new File.dirname(\`node --print "require.resolve('react-native-maps/package.json')"\`)).relative_path_from(Pathname.new(Dir.pwd)).to_s`,
     anchor: /use_native_modules/,
     offset: 0,
     comment: '#'
```

which results in `../../../node_modules/......` instead of `/User/me/Documents/projects/app/node_modules/....` in my podfile.lock

We're using a ton of libraries, however this is the only one we have issues with regarding module resolving

**Comment by salah-ghanim** [comment]
@LeunensMichiel I think this is the wrong repo to submit this feature / pull request right ? it seems like expo's map plugin have a mistake in their pod linking.

**Reply by LeunensMichiel** [reply]
You're right, I will move this upstream!

Btw, thank you for your ongoing work on the new architecture update 💪

**Reply by mieszko4** [reply]
Is there a link to issue on expo's map plugin?
